verilog/formatting/tree_unwrapper.cc

```diff
@@ -3123,6 +3131,36 @@ void TreeUnwrapper::ReshapeTokenPartitions(
       break;
     }
 
+    case NodeEnum::kRegisterVariable:
+    case NodeEnum::kInstantiationType: {
+      size_t partition_size = partition.Children().size();
+      if (partition_size >= 4) {
+        auto subnode = verible::GetSubtreeAsNode(node, tag, 0);
+        if (!subnode) {
+          subnode = verible::GetSubtreeAsNode(node, tag, 1);
+          if (!subnode || subnode->Tag().tag !=
+                              static_cast<int>(NodeEnum::kUnpackedDimensions)) {
+            break;
+          }
+        } else {
+          subnode = verible::GetSubtreeAsNode(*subnode, subnode->Tag().tag, 3);
+          if (!subnode || subnode->Tag().tag !=
+                              static_cast<int>(NodeEnum::kPackedDimensions)) {
+            break;
+          }
+        }
+        auto& colon = partition.Children()[partition_size - 3];
+        if (colon.Value().TokensRange().begin()->Text() != ":") break;
+        AttachSeparatorToPreviousOrNextPartition(&colon);
+        verible::MergeLeafIntoPreviousLeaf(
+            &partition.Children()[partition_size - 2]);
+        verible::FlattenOneChild(partition, partition_size - 3);
+        verible::MergeLeafIntoPreviousLeaf(
+            &partition.Children()[partition_size - 2]);
```

Comment:
I trust that this does the job but it's not obvious to me why we're doing this. I mean I kind of get it from the PR description but personally I prefer most of the necessary info directly in the code.